kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/metadata.kt

```diff
@@ -486,13 +487,17 @@ internal fun TargetProperty.generator(
   )
 }
 
-private fun List<AnnotationSpec>?.qualifiers(elements: Elements): Set<AnnotationSpec> {
+private fun List<AnnotationSpec>?.qualifiers(
+  messager: Messager,
+  elements: Elements
+): Set<AnnotationSpec> {
   if (this == null) return setOf()
   return filterTo(mutableSetOf()) {
-    val typeElement = checkNotNull(elements.getTypeElement(it.typeName.rawType().canonicalName)) {
-      "Could not get the type element of $it"
+    val typeElement: TypeElement? = elements.getTypeElement(it.typeName.rawType().canonicalName)
```

Comment:
Wouldn't `typeElement` here represent the enclosing type, and not an annotation on the element we're looking at (as in the example with `@SuppressLint`)? I think a test would help visualize the problem that's being solved.

Reply:
I can't really reproduce it as it's specific to how AGP and laptop wire their compilation classpaths. This is iterating over all the annotations on a given property, so the element would be the annotation type